src/generic/UnivPoly.jl

```diff
@@ -958,8 +942,7 @@ end
 _change_univ_poly_ring(R, Rx, cached::Bool) = universal_polynomial_ring(R, symbols(Rx); internal_ordering=internal_ordering(Rx), cached)[1]
 
 function change_base_ring(R::Ring, p::UnivPoly{T}; cached::Bool=true, parent::UniversalPolyRing = _change_univ_poly_ring(R, parent(p), cached)) where {T <: RingElement}
-   base_ring(parent) != R && error("Base rings do not match.")
-   return _map(R, p, parent)
+   return UnivPoly(change_base_ring(R, upgrade(AbstractAlgebra.parent(p), data(p)); parent = mpoly_ring(parent)), parent)
```

Comment:
How about introducing a new helper, like this (untested code)
```
function upgrade(p::UniversalPolyRingElem)
  return upgrade(parent(p), data(p))
end
```
then calls to `upgrade` can further be simplified, e.g. here
```suggestion
   return UnivPoly(change_base_ring(R, upgrade(p); parent = mpoly_ring(parent)), parent)
```

Reply:
We can add this new `upgrade` method but #2198 changes this anyway.